src/rospkg/rospack.py

```diff
@@ -362,9 +362,18 @@ def _implicit_rosdeps(self, package):
         self._rosdeps_cache[package] = s = set()
 
         # take the union of all dependencies
-        packages = self.get_depends(package, implicit=True)
-        for p in packages:
-            s.update(self.get_rosdeps(p, implicit=False))
+        packages = []
+        try:
+            packages = self.get_depends(package, implicit=True)
+        except ResourceNotFound as e:
+            del self._rosdeps_cache[package]
+            packages = e.get_depends()
+        if packages:
```

Comment:
This patch doesn't instantiate a `ResourceNotFound` with the new arguments anywhere so this case will never happen - `get_depends` will always be empty.